src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,178 @@
+package duke.task;
+
+import duke.parser.MissingInputFieldException;
+
+/**
+ * Represents a task.
+ */
+public abstract class Task {
+    private boolean isDone;
+    protected TaskType type;
+    public static String delimiter;
+    public static String command;
+    protected static String dataStringSplitter = " \\| ";
+    // temporary measure before storage related methods are migrated over to the storage class
+    protected static String storageDataStringSplitter = " | ";
```

Comment:
I like how you try to avoid 'too short' names to explain what the variable does. It's not excessively lengthy, so it only helps me understand your code better.